android/app/src/main/java/fi/fmi/mobileweather/UpdateWidgetJobIntentService.java

```diff
@@ -166,16 +171,37 @@ protected void updateCoordinates(double latitude, double longitude, int widget_i
         latitude = (double)Math.round(latitude * 10000d) / 10000d;
         longitude = (double)Math.round(longitude * 10000d) / 10000d;
 
-        String params = "locations="+latitude+","+longitude+"&l="+language
-                +"&version="+version;
+        String latlon = latitude+","+longitude;
+        /*String params = "locations="+latitude+","+longitude+"&l="+language
+                +"&version="+version;*/
 
         //Log.d("params", params);
 
         // Large widget
 
-        String url =  "https://widget.weatherproof.fi/android/androidwidget.php?"+params;
+        // forecast url
+        String url = "https://data.fmi.fi/fmi-apikey/ff22323b-ac44-486c-887c-3fb6ddf1116c/timeseries?latlon=" +
+                latlon +
+                "&endtime=data&format=json&attributes=geoid&lang=" +
+                language +
+                "&tz=utc&who=MobileWeather&producer=default&param=geoid,epochtime,localtime,utctime,name,region,iso2,sunrise,sunset,sunriseToday,sunsetToday,dayLength,modtime,dark,temperature,feelsLike,dewPoint,smartSymbol,windDirection,windSpeedMS,pop,hourlymaximumgust,relativeHumidity,pressure,precipitation1h,windCompass8";
+        Log.d("DownloadData json", "url with coordinates: " + url);
+
+//        String url = "https://data.fmi.fi/fmi-apikey/ff22323b-ac44-486c-887c-3fb6ddf1116c/timeseries?param=geoid,name,region,latitude,longitude,region,country,iso2,localtz&latlon=62.5,26.2&format=json";
+
+        // String url =  "https://widget.weatherproof.fi/android/androidwidget.php?"+params;
+
+        String url2 = "";
+        if (!version.equals("classic")) {
+            // observations url (with 1 station)
+            url2 = "https://data.fmi.fi/fmi-apikey/ff22323b-ac44-486c-887c-3fb6ddf1116c/timeseries?latlon=" +
```

Comment:
Same here remove the url with apikey. I think that we could remove whole observation query, because new widgets dont't use observations.

Reply:
UpdateWidgetJobIntentService changes can be ignored because that file will be removed soon. :)

Reply:
File removed